docs/tutorials/telescope_prog_testing.md

```diff
@@ -0,0 +1,126 @@
+# Testing
+
+The Academic Observatory provides a framework that aids in unit and functional testing.  The tests are found in the `tests` folder. Telescope tests are found in `tests/observatory/dags/telescopes`.
+
+Academic Observatory uses the `unittest` Python framework as a base, and provides additional methods to run tasks and test DAG structure.  It also uses the Python `coverage` package to analyse test coverage.
+
+Test methods can be found in the module `observatory.platform.utils.test_utils`.
+
+To take advantage of the framework, instead of inheriting `unittest.TestCase` in test classes, each test class inherits `ObservatoryTestCase`.
+
+## Testing DAGs
```

Comment:
Maybe split it out into three main subheadings under `Testing`:

* Testing DAG structure
* Testing DAG loading
* Functional testing